src/nvim/drawline.c

```diff
@@ -2110,21 +2106,34 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
       }
       ptr++;
 
+      decor_attr = 0;
       if (extra_check) {
         bool no_plain_buffer = (wp->w_s->b_p_spo_flags & SPO_NPBUFFER) != 0;
         bool can_spell = !no_plain_buffer;
 
-        // Get syntax attribute, unless still at the start of the line
+        // Get extmark and syntax attributes, unless still at the start of the line
         // (double-wide char that doesn't fit).
         v = (ptr - line);
+        if (has_decor && v > 0) {
+          decor_attr = extmark_attr;
+
+          decor_conceal = decor_state.conceal;
+          if (decor_conceal && decor_state.conceal_char) {
+            decor_conceal = 2;  // really??
+          }
+
+          can_spell = TRISTATE_TO_BOOL(decor_state.spell, can_spell);
```

Comment:
This change from #23726 works to fix the failing test, without breaking others...
```diff
@@ -2121,8 +2121,6 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
           if (decor_conceal && decor_state.conceal_char) {
             decor_conceal = 2;  // really??
           }
-
-          can_spell = TRISTATE_TO_BOOL(decor_state.spell, can_spell);
         }

         if (has_syntax && v > 0) {
@@ -2183,7 +2181,8 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool nochange,
         v = (ptr - line);
         if (has_spell && v >= word_end && v > cur_checked_col) {
           spell_attr = 0;
-          if (c != 0 && ((!has_syntax && !no_plain_buffer) || can_spell)) {
+          if (c != 0 && ((!has_syntax && can_spell) ||
+                         TRISTATE_TO_BOOL(decor_state.spell, can_spell))) {
             char *prev_ptr;
             char *p;
             int len;
```